### ISSUE TITLE
Importing Fabric mutates global warnings state

### ISSUE THREAD

**Issue by Julian** [fabric/fabric#1602]
`fabric.network` mutates the global warnings state:

https://github.com/fabric/fabric/blob/0a3d11f580e6ccd274727d938dbca0f191d911bb/fabric/network.py

This causes any module that imports fabric (which in turn ends up importing `fabric.network`) to have all of any deprecation warnings, including those unrelated to fabric or its paramiko import, to be hidden.

The filter there should be in a with block to restore the global state afterwards.

**Comment by bitprophet** [comment]
Probably worth digging up why that was added in the first place, it's entirely possible it stems from some older Python versions that are no longer widely used. If it seems still useful to have, then yea, agree it'd be nice to only apply it when specifically needed and undo it after.

**Comment by Julian** [comment]
Yeah, https://github.com/fabric/fabric/commit/5b24075966b15b414ce537480e1a71d85fea74bb appears to be the commit, which I'm guessing is probably something having to do with paramiko emitting a warning telling people Python 2.6 is dead.

Does fabric still support 2.6?

**Comment by bitprophet** [comment]
Yes, we do still support Python 2.6 (this won't change except in 2.0 or 3.0), but I also now maintain Paramiko, so if it's still emitting those warnings itself, it's not something I was aware of and not sure I want it.

Looked and cannot find any use of `warnings` (besides for unknown host keys, at runtime, which is clearly distinct) nor mention of `DeprecationWarning` in Paramiko's codebase on either the 2.x or 1.18 branches. (Or, for that matter, via `git log -G<term>`...)

I just attempted to recreate by commenting out the `warnings.simplefilter` line and doing `python -c "import fabric.network"` under Python 2.6.9, and, no warnings. Again, same on both Paramiko 1.18 (which imports PyCrypto) and 2.1 (which imports Cryptography). My _guess_ is maybe it was coming out of some older version of PyCrypto back in the day?

Either way I think it might be safe to take the axe to this chunk of code; can always reinstate (with a smarter context manager approach, as noted) if the warnings pop up again for people.
